src/main/java/org/wildfly/security/auth/realm/token/TokenSecurityRealm.java

```diff
@@ -211,6 +226,11 @@ public Builder principalClaimName(String name) {
             return this;
         }
 
+        public Builder principalClaimName(Function<Attributes, String> func) {
```

Comment:
I guess if no function is given in the builder then it should default to a function which uses the specified claim name.  Otherwise I think that value ends up unused.

Reply:
But if one does specify the function, then the principalClaimName would end up not being used. If you're wanting that the getRealmIdentityPrincipal() impl simply calls the function to obtain a principal, and default the function to the current logic that makes sense.

Reply:
See the current changes.